policy/lib/tekton/trusted.rego

```diff
@@ -51,6 +51,12 @@ _newer_record_exists(task) if {
 
 	newest_record := time_lib.newest(records)
 	newest_record.ref != ref.pinned_ref
+
+	# newest record could have the same effective_on as the record for the given
+	# task, in that case we can't claim that the newer record exists
+	some record in records
```

Comment:
I wonder if this should be handled in `time_lib.newest(records)`

Reply:
I think I tried that and gave up, not sure why, perhaps many tests failed?